### DISCUSSION TITLE
Migrate Dev - Was Modified After it was Applied

### DISCUSSION THREAD

**Discussion by Etep15** [prisma/prisma#19226] [Q&A]
So I ran into this issue after I had a failed migration. 

I resolved most of it but I can't get past a migrate dev. I see this errror.

The migration `MY_MIGRATION` was modified after it was applied.

And then it wants to reset my DB. Which I do not want to do with all the data I have in there.

The steps I did to get here are :

1. prisma migrate dev
2. _migration failed_
3. edited the migration file (I think this is where the issue stems from)
4. prisma migrate resolve --rolled-back "MY_MIGRATION"
5. edited the migration file again to confirm changes
6. prisma migrate deploy
7. _successful deployment and DB reflecting changes_
8. made a small change (varchar(20) to varchar(24) in prisma.schema just to ensure everything was back on track
9. npx prisma migrate dev
10. _The migration `MY_MIGRATION` was modified after it was applied._
11. prisma migrate resolve --applied "MY_MIGRATION"
12. _The migration `MY_MIGRATION` is already recorded as applied in the database._

So I'm stuck. It says everything is up to date but something prisma is looking at is saying no. Is there a way to force it to mark things as resolved?

**Comment by ludralph** [comment]
Hi @Etep15 👋 

Can you please take a look at [Prisma migrate diff and DB execute](https://www.prisma.io/docs/guides/migrate/production-troubleshooting#fixing-failed-migrations-with-migrate-diff-and-db-execute) command to understand what changed and how to fix it?

**Reply by nurul3101** [reply]
@Etep15 Did you had a chance to look at Raphael's answer?

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏

**Comment by garlab** [comment]
I had the same problem after a failed migration that I manually fixed. I [followed this tutorial](https://echobind.com/post/make-prisma-ignore-a-migration-change) and updated the hash of the migration file in the migration table and it worked.

**Reply by ronniechoyy** [reply]
Thanks, this one save my life

**Reply by erictu22** [reply]
we just deleted that migration row from the `_prisma_migrations` table and reran `prisma migrate dev` without any issues

**Comment by larinius** [comment]
So how to fix that? I changed migration code manually and it worked fine. until Prisma desided to throw error 
The migration `XYZ` was modified after it was applied. - How to disable this feature at all, it is only pain, it never solves anything. only creates trouble, Default Prisma migration script is dumb as hell so let us fix it at least.

The most hilarious part - Prisma suggests  RESET  Schema.. What person in sane mind added this "feature"? Reset schema on running project, seriously? If we could reset we don't need migrations at all, we could just wipe our DB and apply all on empty DB..
